[Feat] Conditional Notifications for custom actions (#1574)
mckinsey/vizro: vizro-core/docs/pages/user-guides/custom-actions.md

    === "Result"

        ![](../../assets/user_guides/actions/custom-actions4.png)

## Notifications

> You can display notifications to show the progress and outcome of a custom action using the built-in  `"progress"`, `"success"` and `"error"` keys in the action's `notifications` argument:

Reviewer: ```suggestion
You can use notifications to show the progress and outcome of a custom action with the built-in  `"progress"`, `"success"` and `"error"` keys in the action's `notifications` argument:
```

        pages:
          - components:
              - type: button
                text: Run pipeline
                actions:
                  - type: action
                    function:
                      _target_: __main__.run_pipeline
                    outputs: pipeline_output
                    notifications:
                      progress: Running pipeline...
                      success: Pipeline completed.
                      error: Pipeline failed.
                      pipeline_partial_success: Pipeline partially completed.
                      pipeline_partial_error: Pipeline partially failed.
              - type: text
                id: pipeline_output
                text: Click the button to run the pipeline.
            layout:
              type: flex
            title: Custom notification keys
        ```

    === "Result"

        [![CustomNotificationKeys]][customnotificationkeys]

### Dynamic notification content

Notification messages can include _template variables_ that are filled with values from the action at runtime. This enables you to display dynamic information, such as results or error details, directly in the notification.

> Two templates are supported to make it possible to provide more informative and contextual feedback to users without hardcoding the message content:

Reviewer: ```suggestion
There are two templates to provide more informative and contextual feedback to users without hardcoding the message content:
```
